File: objectivec-spritekit/Endless Runner/Endless Runner/MyScene.m
//
//  MyScene.m
//  Endless Runner
//
//  Created by Christopher Caleb on 14/04/2014.
//  Copyright (c) 2014 Christopher Caleb. All rights reserved.
//

#import "MyScene.h"
#import "Scroller.h"

@implementation MyScene
{
    SKTextureAtlas *textureAtlas;
    Scroller *scroller;
    CGFloat scrollSpeed;
}

-(id)initWithSize:(CGSize)size {
    if (self = [super initWithSize:size]) {
        self.backgroundColor = [SKColor colorWithRed:0.15 green:0.15 blue:0.3 alpha:1.0];
        
        textureAtlas = [SKTextureAtlas atlasNamed:@"wallslices"];
        scroller = [[Scroller alloc] initWithNode:self withTextureAtlas:textureAtlas];
        scrollSpeed = MIN_SCROLL_SPEED;
    }
    return self;
}

-(void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
    /* Called when a touch begins */
}

-(void)update:(CFTimeInterval)currentTime {
    [scroller moveViewportXBy:scrollSpeed];
    scrollSpeed += SCROLL_ACCELERATION;
    if (scrollSpeed > MAX_SCROLL_SPEED)
    {
        scrollSpeed = MAX_SCROLL_SPEED;
    }
}

@end

File: swift-spritekit/Endless-Runner/MyScene.swift
//
//  MyScene.swift
//  Endless-Runner
//
//  Created by Christopher Caleb on 12/06/2014.
//  Copyright (c) 2014 Christopher Caleb. All rights reserved.
//

import Foundation
import SpriteKit

class MyScene: SKScene {
    
    let MIN_SCROLL_SPEED:    CGFloat = 5.0
    let MAX_SCROLL_SPEED:    CGFloat = 15.0
    let SCROLL_ACCELERATION: CGFloat = 0.005
    
    let _textureAtlas: SKTextureAtlas
    let _scroller: Scroller?
    var _scrollSpeed :CGFloat
    var _viewportX :CGFloat
    
    init(size: CGSize) {
        _viewportX = 0.0
        _scrollSpeed = MIN_SCROLL_SPEED
        _textureAtlas = SKTextureAtlas(named: "wallslices")
        super.init(size: size)
        
        _scroller = Scroller(node: self, textureAtlas: _textureAtlas)
        addChild(_scroller)
    }
    
    override func update(currentTime: NSTimeInterval) {
        _scrollSpeed += SCROLL_ACCELERATION
        if _scrollSpeed > MAX_SCROLL_SPEED {
            _scrollSpeed = MAX_SCROLL_SPEED
        }
        _scroller!.moveViewportXBy(_scrollSpeed);
    }
}